packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -527,7 +527,7 @@ export abstract class FluidDataStoreContext
 	createChildDataStore<T extends IFluidDataStoreFactory>(
 		childFactory: T,
 	): ReturnType<Exclude<T["createDataStore"], undefined>> {
-		const maybe = this.registry?.get(childFactory.type);
+		const maybe = this.registry?.getSync?.(childFactory.type);
 
 		const isUndefined = maybe === undefined;
 		const isPromise = isPromiseLike(maybe);
```

Comment:
Should never be a promise now, right? Does this need to try `get` if `getSync` didn't deliver?

Reply:
No. that was the problem I'm fixing. i tried to overload get to return a promise or not a promise, banking on the await behavior for back compat, here we always want the synchronous results. Now get will always return a promise, and the only way to get a non-promise is to call `getSync`. 

This code can be simplified now, and i'll take that as a follow up.